### PR TITLE
Add self-assignable user role support

### PR DESCRIPTION
**Pull request by dhinakg**

### CODE REVIEW

**Review by Erisa — CHANGES_REQUESTED** [review]
Discussed on Discord, needs handling for `UserRoles` being null in config, and I recommend creating an `ExecuteCheck` for that purpose.

Currently crashes if the user doesn't add the new config values.

I like it though! Code quality is great.

File: Modules/UserRoles.cs [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+
+        [
+            Command("leave-insiders"),
+            Description("Removes you from Insider roles"),
+            Aliases("leave-insider"),
+            HomeServer
+        ]
+        public async Task LeaveInsiders(CommandContext ctx)
+        {
+            await UserRoles.RemoveUserRolesAsync(ctx, x => x.Name.Contains("Insider"));
```
Comment (Erisa) [inline Modules/UserRoles.cs:143]: This isn't the most elegant way of handling this, I would recommend iterating through the list of IDs in the config.

**Review by Erisa — CHANGES_REQUESTED** [review]
Apologies for the delay in re-reviewing this, if you can resolve the one comment I made then I'll be more than happy to merge.

Thank you!

File: Modules/UserRoles.cs [hunk 2]
```diff
@@ -74,8 +74,6 @@ public static async Task RemoveUserRolesAsync(CommandContext ctx, Func<System.Re
                 DiscordRole roleToGrant = guild.GetRole((ulong)roleId.GetValue(Program.cfgjson.UserRoles, null));
                 await ctx.Member.RevokeRoleAsync(roleToGrant);
             }
-
-            await ctx.Message.DeleteAsync();
```
Comment (dhinakg) [inline Modules/UserRoles.cs]: Attempting to delete a message multiple times causes an exception

Comment (Erisa) [reply Modules/UserRoles.cs]: Maybe you could try/catch it? Failing to delete the message isn't an exception worth acting on after all, can just be discarded.

**Pull request merged**